docs/advanced/AsyncActions.md

```diff
@@ -293,79 +293,16 @@ Remember that reducers are just functions, so you can use functional composition
-import { selectReddit, requestPosts, receivePosts } from './actions';
-import rootReducer from './reducers';
-
-const store = createStore(rootReducer);
-
-store.dispatch(selectReddit('reactjs'));
-
-store.dispatch(requestPosts('reactjs'));
-fetch(`http://www.reddit.com/r/${reddit}.json`)
-  .then(response => response.json())
-  .then(json =>
-    store.dispatch(receivePosts(reddit, json))
-  )
-  .then(() => {
-    console.log(store.getState());
-  });
-```
-
->##### Note on `fetch`
-
->We use [`fetch` API](https://developer.mozilla.org/en/docs/Web/API/Fetch_API) in the examples. It is a new API for making network requests that replaces `XMLHttpRequest` for most common needs. Because most browsers don’t yet support it natively, we suggest that you use [`isomorphic-fetch`](https://github.com/matthew-andrews/isomorphic-fetch) library:
-
->```js
-// Do this in every file where you use `fetch`
->import fetch from 'isomorphic-fetch';
->```
-
->Internally, it uses [`whatwg-fetch` polyfill](https://github.com/github/fetch) on the client, and [`node-fetch`](https://github.com/bitinn/node-fetch) on the server, so you won’t need to change API calls if you change your app to be [universal](https://medium.com/@mjackson/universal-javascript-4761051b7ae9).
-
->Be aware that any `fetch` polyfill assumes a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) polyfill is already present. The easiest way to ensure you have a Promise polyfill is to enable Babel’s ES6 polyfill in your entry point before any other code runs:
-
->```js
->// Do this once before any other code in your app
->import 'babel-core/polyfill';
->```
-
-We can call the APIs and dispatch the relevant actions from our components, too. However, it quickly gets tedious. Usually you want some kind of common logic before performing a request, such as looking up something in the state, and maybe deciding not to fetch because the data is cached.
-
-Clearly, actions can’t express control flow. The best tool for control flow is a function. A function can have an `if` statement, or an early `return`. If a function has access to a `dispatch` method, it can call it many times, potentially asynchronously. Does this ring a bell?
+Finally, how do we use the synchronous action creators we [defined earlier](#synchronous-action-creators) together with network requests? The standard
+ way to do it with Redux is to use the [edux-thunk middleware](https://github.com/gaearon/redux-thunk). We'll explain how middleware works in general [later](Middleware.md); for now, there 
+ is just one important thing you need to know: by using this specific middleware, an action creator can return a function instead an action object.
+ This way, the function creator becomes a [thunk](https://en.wikipedia.org/wiki/Thunk).
+ 
+When a function creator returns a function, that function will get executed by the redux-thunk middleware. This function doesn't need to be pure; 
```

Comment:
I just think "redux-thunk middleware" doesn't look nicely in text. IMO it should either be a link, or "Redux Thunk middleware".